src/node/channels.h

```diff
@@ -631,24 +681,29 @@ namespace ccf
           outgoing_msg->raw_cipher);
         outgoing_msg.reset();
       }
-
-      auto node_cv = make_verifier(node_cert);
-      LOG_INFO_FMT("Node channel with {} is now established.", peer_id);
-
-      LOG_TRACE_FMT(
-        "Node certificate serial numbers: node={} peer={}",
-        node_cv->serial_number(),
-        peer_cv->serial_number());
     }
 
     void initiate()
     {
-      if (status == WAITING_FOR_FINAL || status == ESTABLISHED)
+      if (status == WAITING_FOR_FINAL)
         return;
 
       LOG_INFO_FMT("Initiating node channel with {}.", peer_id);
 
-      status = INITIATED;
+      key_exchange_in_progress = true;
+
+      if (status != ESTABLISHED)
+        status = INITIATED;
```

Comment:
```suggestion
      }
        status = INITIATED;
      }
```

Reply:
Awww, I thought I caught all of them this time!